cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -169,12 +179,45 @@ func (p *BicepProvider) Plan(
 				return
 			}
 
-			target := infra.NewSubscriptionDeployment(
-				p.azCli,
-				p.env.GetLocation(),
-				p.env.GetSubscriptionId(),
-				fmt.Sprintf("%s-%d", p.env.GetEnvName(), time.Now().Unix()),
-			)
+			var target infra.Deployment
+
+			switch template.TargetScope() {
+			case azure.DeploymentScopeSubscription:
+				target = infra.NewSubscriptionDeployment(
+					p.azCli,
+					p.env.GetLocation(),
+					p.env.GetSubscriptionId(),
+					fmt.Sprintf("%s-%d", p.env.GetEnvName(), time.Now().Unix()),
+				)
+			case azure.DeploymentScopeResourceGroup:
+				if !p.alphaFeatureManager.IsEnabled(ResourceGroupDeploymentFeature) {
+					asyncContext.SetError(ErrResourceGroupScopeNotSupported)
+					return
+				}
+
+				if alpha.ShouldWarn(ResourceGroupDeploymentFeature) {
+					p.console.MessageUxItem(ctx, alpha.WarningMessage(ResourceGroupDeploymentFeature))
+				}
+
+				if p.env.Getenv(environment.ResourceGroupEnvVarName) == "" {
+					asyncContext.SetError(
+						fmt.Errorf(
+							"%s must be set to the name of the resource group to use",
+							environment.ResourceGroupEnvVarName,
+						),
+					)
+					return
+				}
+
+				target = infra.NewResourceGroupDeployment(
+					p.azCli,
+					p.env.GetSubscriptionId(),
+					p.env.Getenv(environment.ResourceGroupEnvVarName),
+					fmt.Sprintf("%s-%d", p.env.GetEnvName(), time.Now().Unix()),
+				)
+			default:
+				panic(fmt.Sprintf("unknown template target scope: %s", template.TargetScope()))
+			}
```

Comment:
I wanted to (and when taking a look at the diff before sending it off actually tried to do so), but realized that `scopeForTemplate()` is different in that it returns an `infra.Scope` whereas here we want an `infra.Deployment`.

While there is only a single caller, do you think it would help to extract this into `scopeForDeployment` and place it near `scopeForTemplate` in the file itself?

Reply:
Keeping this as is for now, @wbreza, but happy to go make improvements here in a follow up if you have a better idea how how we want to structure this.